### ISSUE TITLE
Getting device list

### ISSUE THREAD

**Issue by seemmichael** [aromka/myqcontroller#1]
I have followed the readme on both Mac and raspberry pi, and the console stops on "Getting device list..." every time.  I have refreshed/reinstalled/deleted/etc, but I cannot get past this step.

**Comment by BoPeng** [comment]
I am in the same situation. The node server says

```
=== === === MyQ Controller === === ===
[04/04 21:43:58] Retrieved config for server: 192.168.0.170:39500
[04/04 21:43:59] Config loaded
[04/04 21:46:43] Handling request for: /ping
[04/04 21:46:43] Getting ping... replying pong
[04/04 21:47:04] Handling request for: /ping
[04/04 21:47:04] Getting ping... replying pong
[04/04 21:47:10] Handling request for: /init
[04/04 21:47:10] Received init request
[04/04 21:47:10] Initializing...
[04/04 21:47:10] Getting device list...
[04/04 21:47:10] Handling request for: /init
[04/04 21:47:10] Received init request
[04/04 21:47:10] Initializing...
[04/04 21:47:10] Getting device list...
```

and no device is added to `My Things`. Note that I have a driveway gate in myQ, but it should be recognized as a garage door opener AFAIK.

**Comment by BoPeng** [comment]
Checking [the code](https://github.com/aromka/myqcontroller/blob/master/server/service/myq.js#L145), it appears that it only works for garage door and switch, not gate.... Not sure if gate can be easily supported. Note that MyQ Lite's garage door opener device can be used for my gate.

**Comment by gwilli** [comment]
I also have the same issue. @seemmichael and @BoPeng when you went to the `My Hubs` section in SmartThings IDE, were you able to see a port number? There's no port number listed for my hub so I left it as the default. I wonder if that has something to do with it?

**Comment by BoPeng** [comment]
No. The default port works. You will see some error message if you change to another port. I could actually add some `console.log(...)` statements and see where the code went wrong but making the whole thing work would be extremely time consuming.

So in the end I switched back to myQ Lite.

**Comment by gwilli** [comment]
@BoPeng Thanks for the suggestions. I was able to debug it and fix the issue _for my garage door_. It's working now for me.

The problem is an age old problem: how to match on string data? Line 138 of `myq.js` is looking at the `MyQDeviceTypeName` and attempting to replace all or part of the string containing `VGDO` with `GarageDoorOpener`.  For my garage door, the garage door device from the device list has the name `Garage Door Opener WGDO`. I'd guess that for @aromka's garage door, the value of `MyQDeviceTypeName` is just `VGDO` and that worked. Looks like MyQ/Chamberlain/LiftMaster keeps changing it up to keep this app from working.

I might see if I can come up with a better way of identifying the garage door opener from the device list response and submit a pull request. We'll see. Thanks again for pointing me in the right direction.

**Comment by BoPeng** [comment]
My problem is a bit more complicated as I have a lift master driveway gate. MyQ lite's garage door device works for me but this one does not. I might be able to make it work by comparing the code against MyQ lite but I do not have time for that now. I am also a bit discouraged on the node server side because it is difficult to run the server automatically on my qnap NAS.

**Comment by perpetualbliss84** [comment]
This just started happening to me out of the blue the other day. Cannot get it to work at all. has anyone sorted this out? Thanks.

@aromka Any idea why this would have stopped working suddenly? SmartApp communicates with the server and such but just sitting at getting device list.

**Comment by nickfoti** [comment]
@perpetualbliss84 I've been having the same issue for months but finally had a chance to take a look at why today. From the looks of it, the https://myqexternal.myqdevice.com/api/v4/userdevicedetails/get call is returning unauthorized:

{
  "ReturnCode": "216",
  "ErrorMessage": "Unauthorized (216)",
  "CorrelationId": "244827af-610a-4e3e-865e-8627f4b60286"
}

Perhaps the AppID is no longer valid or has been blocked?

**Comment by perpetualbliss84** [comment]
I kinda that same conclusion. Doesn't seem like the app is being supported anymore. I changed to something else for the time being.
________________________________

**Comment by nickfoti** [comment]
@perpetualbliss84 It seems like several other projects popped up in the last year using the same AppID. I'm unclear on how you obtain one to begin with, but that could be the reason why it no longer works. Did you switch to MyQ Lite? I'm interested to know what you have currently working.

**Comment by perpetualbliss84** [comment]
Yes, thats what I switched to. Bought a few multipurpose sensors as well just to know if it actually is closed or open. I am hoping in the near future that Chamberlain releases for Smartthings again as they have for almost every other platform.
________________________________

**Comment by nickfoti** [comment]
Gotcha, that would be nice - not holding my breath though. I begrudgingly reinstalled the MyQ app and have been using that since this stopped working.
